src/clients/c++/perf_client/load_manager.cc

```diff
@@ -213,33 +306,297 @@ LoadManager::InitManagerInputs(
   // needed input. We (re)use this buffer for all input values.
   if (max_input_byte_size > 0) {
     if (zero_input) {
-      local_manager->input_buf_.resize(max_input_byte_size, 0);
+      input_buf_.resize(max_input_byte_size, 0);
     } else {
-      local_manager->input_buf_.resize(max_input_byte_size);
-      for (auto& byte : local_manager->input_buf_) {
+      input_buf_.resize(max_input_byte_size);
+      for (auto& byte : input_buf_) {
         byte = rand();
       }
     }
   }
 
   // Similarly, handle the input_string_buf_
   if (needs_string_input) {
-    local_manager->input_string_buf_.resize(max_batch1_num_strings);
+    input_string_buf_.resize(max_batch1_num_strings);
     if (!string_data.empty()) {
       for (size_t i = 0; i < max_batch1_num_strings; i++) {
-        local_manager->input_string_buf_[i] = string_data;
+        input_string_buf_[i] = string_data;
       }
     } else {
       for (size_t i = 0; i < max_batch1_num_strings; i++) {
-        local_manager->input_string_buf_[i] = GetRandomString(string_length);
+        input_string_buf_[i] = GetRandomString(string_length);
       }
     }
   }
 
   // Reserve the required vector space
-  local_manager->threads_stat_.reserve(local_manager->max_threads_);
+  threads_stat_.reserve(max_threads_);
+
+  return nic::Error::Success;
+}
+
+nic::Error
+LoadManager::InitSharedMemory()
+{
+  nic::Error err;
+
+  RETURN_IF_ERROR(
+      factory_->CreateSharedMemoryControlContext(&shared_memory_ctx_));
+  std::unique_ptr<nic::InferContext> ctx;
+  RETURN_IF_ERROR(factory_->CreateInferContext(&ctx));
 
-  *manager = std::move(local_manager);
+  // Allocate the shared memory for outputs
+  for (const auto& output : ctx->Outputs()) {
+    int64_t batch1_bytesize = ctx->ByteSize(output->Dims(), output->DType());
+    if (batch1_bytesize < 0) {
+      batch1_bytesize = output_shm_size_;
+    }
+    uint8_t* output_shm_ptr;
+    size_t alloc_size = batch1_bytesize * batch_size_;
+    if (shared_memory_type_ == SharedMemoryType::SYSTEM_SHARED_MEMORY) {
+      std::string shm_key("/" + output->Name());
+      int shm_fd_op;
+      RETURN_IF_ERROR(
+          CreateSharedMemoryRegion(shm_key, alloc_size, &shm_fd_op));
+      RETURN_IF_ERROR(
+          MapSharedMemory(shm_fd_op, 0, alloc_size, (void**)&output_shm_ptr));
+
+      io_shm_size_[output->Name()] = alloc_size;
```

Comment:
You can use a single map for size and address

Reply:
Done